var/spack/repos/builtin/packages/py-scipy/package.py

```diff
@@ -129,8 +145,37 @@ def setup_build_environment(self, env):
             if self.spec.satisfies("^py-numpy@1.16:1.17"):
                 env.set("NPY_DISTUTILS_APPEND_FLAGS", "1")
 
-        # Pick up Blas/Lapack from numpy
-        self.spec["py-numpy"].package.setup_build_environment(env)
+        # Pick up BLAS/LAPACK from numpy
+        if self.spec.satisfies("@:1.8"):
+            self.spec["py-numpy"].package.setup_build_environment(env)
+
+    # TODO: remove once pip build supports BLAS/LAPACK specification
+    # https://github.com/FFY00/meson-python/pull/122
+    @when("@1.9:")
+    def install(self, spec, prefix):
+        args = [
+            "setup",
+            "build",
+            "-Dblas=" + spec["blas"].libs.names[0],
+            "-Dlapack=" + spec["lapack"].libs.names[0],
+            "--prefix=" + join_path(os.getcwd(), "build-install"),
+            "-Ddebug=false",
+            "-Doptimization=2",
+        ]
+        meson = which("meson")
+        meson(*args)
+        args = [
+            "-m",
+            "build",
+            "--wheel",
+            "-Cbuilddir=build",
+            "--no-isolation",
+            "--skip-dependency-check",
+            ".",
+        ]
+        python(*args)
+        args = std_pip_args + ["--prefix=" + prefix, glob.glob(join_path("dist", "scipy*.whl"))[0]]
+        pip(*args)
```

Comment:
This doesn't seem to be working:
```
ERROR: scipy-1.9.0-cp39-cp39-macosx_12_arm64.whl is not a supported wheel on this platform.
```
I'm not sure why it would complain about that since I'm on macOS 12.5 (arm64).

Reply:
That rings a bell, it's a `packaging` bug and that is a library vendored by `pip`: https://github.com/pypa/packaging/issues/578. There's nothing wrong with the wheel probably, just renaming it `12` -> `12_0` should make `pip` be able to install it.

Reply:
Bleh, I'm not sure how I would even parse the 12 out of the filename in order to replace it with 12_0. Why don't I see this issue with `pip install .`? Is this bug only in wheel filenames created by `build`?

Reply:
Hmm, good question. I determined in https://github.com/pypa/packaging/issues/578#issuecomment-1192291127 that it depends on the Python build you are using. I don't think there should be a difference in wheel build tags between `build` and `pip`; the name is determined by `meson-python`. Maybe an outdated `meson-python` version? It already has a workaround for this: https://github.com/FFY00/meson-python/blob/fc29c1bc820f4b4abbb929aa68d90cf6e6b40a0c/mesonpy/__init__.py#L677-L686. That should be present in `v0.8.0`: https://github.com/FFY00/meson-python/commit/71cbc44af0464f2daeb4855b9cad35e1eb2e637d

Reply:
Ah perfect, let me try newer meson-python!

Reply:
Bleh, it's pinned to `"meson-python>=0.7.0,<0.8.0"` in 1.9.0 but relaxed in main. Guess I'll relax it too.

Reply:
yes, please do. I missed that pin, should have been loosened.